tests/cli/test_output_option.py

```diff
@@ -17,97 +17,210 @@
 """Test behavior of ``--output`` option."""
 
+import pytest
 import yaml
 
 from renku.models.cwl import CWLClass
 
 
-def test_without_an_output_option(runner, client, run):
-    """Test detection of an output file without an output option."""
-    assert 0 == run(args=('run', 'touch', 'hello.txt'))
-    assert 1 == run(args=('run', 'touch', 'hello.txt'))
-    assert 0 == run(args=('run', '--no-output', 'touch', 'hello.txt'))
+@pytest.fixture
+def cli(client, run):
+    """
+    Return a function that runs the provided command and returns the exit code
+    and content of the resulting CWL command line tool.
+    """
 
-    # There should be two command line tool.
-    tools = list(client.workflow_path.glob('*_touch.cwl'))
-    assert 2 == len(tools)
+    def renku_cli(*args):
+        before_cwl_files = set(client.workflow_path.glob('*.cwl'))
+        exit_code = run(args)
+        after_cwl_files = set(client.workflow_path.glob('*.cwl'))
+        new_files = after_cwl_files - before_cwl_files
+        assert len(new_files) <= 1
+        content = read_cwl_file(new_files.pop()) if new_files else None
+        return exit_code, content
 
-    cwls = []
-    for tool_path in tools:
-        with tool_path.open('r') as f:
-            cwls.append(CWLClass.from_cwl(yaml.safe_load(f)))
+    return renku_cli
 
-    assert cwls[0].inputs != cwls[1].inputs
-    assert cwls[0].outputs != cwls[1].outputs
+
+def read_all_cwl_files(client, glob='*.cwl'):
+    """
+    Return an array where its elements are content of CWL file
+    found in the Renku project.
+    """
+    return [read_cwl_file(f) for f in client.workflow_path.glob(glob)]
 
 
-def test_with_an_output_option(runner, client, run):
-    """Test detection of an output file with an output option."""
-    assert 0 == run(args=('run', 'touch', 'hello.txt'))
-    assert 0 == run(
-        args=('run', '--output', 'hello.txt', 'touch', 'hello.txt')
-    )
+def read_cwl_file(cwl_filepath):
+    with cwl_filepath.open('r') as f:
+        return CWLClass.from_cwl(yaml.safe_load(f))
 
-    # There should be two command line tool.
-    tools = list(client.workflow_path.glob('*_touch.cwl'))
-    assert 2 == len(tools)
 
-    cwls = []
-    for tool_path in tools:
-        with tool_path.open('r') as f:
-            cwls.append(CWLClass.from_cwl(yaml.safe_load(f)))
+def test_run_succeeds_normally(cli):
+    """Test when an output is detected"""
+    exit_code, cwl = cli('run', 'touch', 'foo')
 
-    assert cwls[0].inputs == cwls[1].inputs
-    assert cwls[0].outputs == cwls[1].outputs
+    assert exit_code == 0
+    assert len(cwl.inputs) == 1
+    assert cwl.inputs[0].default == 'foo'
+    assert cwl.inputs[0].type == 'string'
+    assert len(cwl.outputs) == 1
+    assert cwl.outputs[0].type == 'File'
+    assert cwl.outputs[0].outputBinding.glob == '$(inputs.input_1)'
 
 
-def test_output_directory_with_output_option(runner, client, run):
-    """Test directory cleanup with the output option."""
```

Comment:
Why remove these tests?

Reply:
This is an undocumented behaviour that existed before and doesn't exist anymore. Before, with --output you can tell Renku to delete a file or directory (that comes after --output). Renku would delete the file or dir and make a commit and then run the command. This was useful to remove an output to make it detectable later.
This was replaced by explicit output hence the behaviour changed and the test is covered in some other tests.